core/state-machine/src/changes_trie/mod.rs

```diff
@@ -126,6 +133,17 @@ impl<'a, H: Hasher, N: BlockNumber> crate::TrieBackendStorage<H> for TrieBackend
 /// Changes trie configuration.
 pub type Configuration = primitives::ChangesTrieConfiguration;
 
+/// Blocks range where configuration has been constant.
+#[derive(Clone)]
+pub struct ConfigurationRange<'a, N> {
+	/// Active configuration.
+	pub config: &'a Configuration,
+	/// Zero block of this configuration. The configuration is active starting from the next block.
+	pub zero: N,
```

Comment:
`Zero` is because currently there's a single possible configuration && it 'starts' at zero block. However, it doesn't actually 'start' at zero - changes trie isn't created for genesis. So it starts from the next block (1). Here's the same situation - if I'd replace it with `start`, then there'll be confusion of when tries are actually started creation. There are good namings for `[begin; end)`, but I need something for `(zero; end]`.

Reply:
Yes 'origin' could fit, but 'zero' is understandable, no worries.